src/main/java/com/es/phoneshop/model/product/ArrayListProductDao.java

```diff
@@ -1,25 +1,71 @@
 package com.es.phoneshop.model.product;
 
+import java.math.BigDecimal;
+import java.util.ArrayList;
+import java.util.Currency;
 import java.util.List;
+import java.util.NoSuchElementException;
+import java.util.stream.Collectors;
 
 public class ArrayListProductDao implements ProductDao {
+    private List<Product> products;
+    private long maxId;
+    private final Object lock = new Object();
+
+    public ArrayListProductDao(){
+        this.products = new ArrayList<>();
+        saveSampleProducts();
+    }
+
     @Override
-    public Product getProduct(Long id) {
-        throw new RuntimeException("Not implemented");
+    public Product getProduct(Long id) throws RuntimeException {
+        return products.stream()
+                .filter(product -> id.equals(product.getId()))
+                .findAny()
+                .orElseThrow(() -> new RuntimeException("Product with id: " + id + " is not found."));
```

Comment:
In previous comment I was talking about NoSuchElementException, not pure RuntimeException, but it's child. Bad practice to use top level exceptions. It's minor change so please do not make new commit, but use amend/squash commit just as example of git facilities. Also please study and use force push git operation.